File: Data Structures & Algorithms/counting-bits/submission-1.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        fin = []
        temp: int = 0
        if n == 0:
            fin.append(0)
        else:
            fin.append(0)
            fin.append(1)
        for i in range(2, n + 1):
            fin.append(fin[i // 2] + (i & 1))
        return fin

File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-1.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        std::vector<int> temp = arr;
        int currMax = -1;
        for (size_t i{0}; i < arr.size(); i++) {
            int ind = (arr.size() - 1) - i;
            arr[ind] = currMax; 
            currMax = (currMax < temp[ind]) ? temp[ind] : currMax;
        }
        return arr;
    }
};